inst/tests/other.Rraw

```diff
@@ -5,26 +5,31 @@ if (exists("test.data.table",.GlobalEnv,inherits=FALSE)) {
   warning("This is dev where with.other.packages should not be run. Instead, use a fresh R session with data.table installed. ",
           "Not doing so in dev can be the cause of both false errors and false passes.")
 }
+if (!"package:data.table" %in% search()) stop("data.table should be already attached. Usage: require(data.table); test.data.table(with.other.packages=TRUE)")
 
 test = data.table:::test
 INT = data.table:::INT
 
-pkgs = c("ggplot2", "hexbin", "plyr", "caret", "xts", "gdata", "zoo", "nlme", "bit64", "knitr", "plm", "parallel", "dtq")
-if (any(duplicated(pkgs))) stop("Internal error: pkgs has a dup")
-for (s in pkgs) {
-  assign(paste0("test_",s), loaded<-suppressWarnings(suppressMessages(require(s, character.only=TRUE))))
-  if (!loaded) cat("\n**** Other package",s,"is not installed. Tests using it will be skipped.\n\n")
-}
+pkgs = c("ggplot2", "hexbin", "plyr", "caret", "xts", "gdata", "zoo", "nlme", "bit64", "knitr", "plm", "parallel")
+if (any(duplicated(pkgs))) stop("Packages defined to be loaded for integration tests in 'inst/tests/other.Rraw' contains duplicates.")
+
+is.require = function(pkg) suppressWarnings(suppressMessages(isTRUE(require(pkg, character.only=TRUE, quietly=TRUE, warn.conflicts=FALSE))))
+loaded = sapply(pkgs, is.require)
+cat("\nPackages to be tested:\n\n")
+print(data.table(pkg=pkgs, loaded)[loaded==TRUE, version:=as.character(sapply(pkg, function(p) format(packageVersion(p))))][])
```

Comment:
This is nice. If a package isn't installed there used to be a message saying its tests will be skipped. Could that message be brought back whilst still keeping the better new table as well?   I imagine that when called from CI, the CI package would catch/warn about packages not installed at that point.  I've more in mind the (much rarer case) of users running `test.data.table(with.other.packages=TRUE)`.

Reply:
@mattdowle Done. I adjusted default to the same way that R CMD check would expect.
By default all integration suggested pkgs are required, this can be turned off by `_R_CHECK_FORCE_SUGGESTS_=FALSE` env var.